sorting/heap_sort_2.c

```diff
@@ -0,0 +1,138 @@
+/**
+ * @file
+ * @author [Dhruv Pasricha](https://github.com/DhruvPasricha)
+ * @brief [Heap Sort](https://en.wikipedia.org/wiki/Heapsort) implementation
```

Comment:
Please add a detailed description of the algorithm.

Reply:
Okay

Reply:
Okay Sir